app/src/service/ocr/index.ts

```diff
@@ -1,35 +1,138 @@
+import { createLogger } from "@/service/factory";
+import { type ParserKeys, parseOcrResult, parsers } from "@/service/ocr/parser";
+import {
+  DocumentImage,
+  DocumentOrientation,
+  orientations,
+  rotateDocumentImage,
+} from "@/utils/document";
 import { createWorker } from "tesseract.js";
 
-import { createLogger } from "../../utils/logger";
-import * as parsers from "./parser";
+export type DocumentMatcher<
+  K extends ParserKeys,
+  P extends Record<string, RegExp> = Record<string, RegExp>
+> = {
+  name: string;
+  id: K;
+  patterns: P;
+};
 
-type OcrOptions = {
+export type OcrOptions = {
   debug: boolean;
 };
 
+export type ProcessedImageResult = {
+  documents: ReturnType<typeof parseOcrResult>;
+  image: DocumentImage;
+  percentages: Record<ParserKeys, number>;
+  confidence: number;
+  rotatedOrientation?: DocumentOrientation;
+};
+
+export type ProcessedRotatedImagesResult = ProcessedImageResult[];
+
+export type TextFromImagePathResult = {
+  text: string;
+  confidence: number;
+};
+
 // main logger
 export const logger = createLogger("ocr-parser", {
-  enabled: process.env.LOG_LEVEL === "debug",
+  enabled: false,
   mixin: () => {
     return { module: "ocr-parser" };
   },
 });
 
-const scanImage = async (path: string, opts: OcrOptions) => {
+const getTextFromImagePath = async (
+  document: DocumentImage,
+  opts: OcrOptions
+) => {
   const worker = await createWorker("eng", 1, {
     logger: opts.debug ? (message) => logger.debug(message) : undefined,
   });
-  const result = await worker.recognize(path);
-  const output = result.data.text;
+
+  const result = await worker.recognize(document.data, {
+    rotateAuto: true,
+  });
+
+  const output = {
+    text: result.data.text,
+    confidence: result.data.confidence,
+  };
+
   await worker.terminate();
 
   return output;
 };
 
+// a function that rotates an image and processes it for each orientation
```

Comment:
So we rotate the image because the user could submit an image of a document that's not in the correct orientation. Tesserect supposedly has the ability to auto-rotate an image, but it didn't work each time I attempted it. Maybe there's an underlying error with the Javascript implementation/wrapper.

Let's say that the image is sent to them via text, but somehow it's not the proper orientation once it's downloaded (I've actually had this happen) and the user isn't sure how to rotate the image. I think most users would try to do this on a computer rather than being aware that they can do so on their mobile device.